### DISCUSSION TITLE
Right Alt issue - two separate letters instead of one keystroke

### DISCUSSION THREAD

**Discussion by BorkowskiDominik** [manna-harbour/miryoku#148] [Q&A] [answered]
Hello,
Thank you for your work. Great layout, but I have small problem.
When I use right alt it does not work as expected for me.
Very often instead of localized letter it is detected as two separated keystrokes.
For example in my case instead of "ć" (right alt + c) i get ".c" (same story for other hand, instead of "ó" [right alt +o] i often get "xo"). I thought that maybe that's problem with my fingers, but I don't have similar issue with home row modifiers like shift etc.
What should I change in sources to make it work fine for me? Probably some timer need to be changed or something like this but I would be gratefull for pointing out where and what to change.

Thanks

**Comment by isavcic** [comment]
I was wondering the same, a quick peek into the QMK config shows that this layout, for reason unbeknownst to me, doesn't use Right Alt at all; all Alts are mapped to Left Alt. @manna-harbour can you shed some light on the reasoning behind this choice?

NB Right Alt is treated as AltGr on Serbian Latin layout, meaning it's actually triggering Ctrl+Alt with one button.

**Reply by manna-harbour** [reply]
See https://github.com/manna-harbour/miryoku/discussions/148#discussioncomment-3911333.

**Comment by manna-harbour** [comment, marked as answer]
Alt is Left Alt on both sides for use as a mod with either hand. Right Alt is below Alt on both sides for use as AltGr with either hand, with appropriate software support on the host. Right Alt has the same mod tap configuration as the home row mods.

@BorkowskiDominik You might not be holding long enough. Try consciously pausing between the hold and the tap. As AltGr isn't on the home row it wont be as susceptible to accidental activation, so there should be some scope to make it easier to activate. You could reduce tapping term, or enable permissive hold. These can be done per key to only apply to AltGr. Right now you'd have to make the changes directly. Support for customising mods will be added in future, see https://github.com/manna-harbour/miryoku/issues/33. It's also possible that AltGr behaves differently from other mods on the host.

**Comment by isavcic** [comment]
> Right Alt is below Alt on both sides for use as AltGr with either hand, with appropriate software support on the host.

I'm sorry, I don't quite follow. There's not a single RALT in miryoku_layer_alternatives.h

**Reply by agisga** [reply]
The keycode used in miryoku_layer_alternatives.h is ALGR_T which is an alias for RALT_T (https://docs.qmk.fm/#/keycodes?id=mod-tap-keys)
